### PR TITLE
Make groupBy trim size configurable at Broker

### PR DESCRIPTION
**Pull request by vvivekiyer**
- Currently, the groupBy trim size for groupByDataTableReducer is determined as `Math.max(limit * 5, DEFAULT_MIN_NUM_GROUPS)`
- DEFAULT_MIN_NUM_GROUPS is hardcoded to 5000 and it's not configurable. 

This PR makes it configurable so that users can control the accuracy of their groupBy queries.

### CODE REVIEW

**Review by SabrinaZhaozyf — APPROVED** [review]
nit: consider renaming to `minBrokerGroupTrimSize` / `pinot.broker.min.broker.group.trim.size` so more clarity and aligning with `min.server.group.trim.size` / `minServerGroupTrimSize`

File: pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BaseReduceService.java [hunk 1]
```diff
@@ -51,12 +51,15 @@ public abstract class BaseReduceService {
   protected final ExecutorService _reduceExecutorService;
   protected final int _maxReduceThreadsPerQuery;
   protected final int _groupByTrimThreshold;
+  protected final int _minGroupByTrimSize;
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BaseReduceService.java:54]: To be consistent with server config name, suggest naming it `_minGroupTrimSize`, same for other places

File: pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java [hunk 2]
```diff
@@ -267,6 +267,8 @@ public static class Broker {
     // used for SQL GROUP BY during broker reduce
     public static final String CONFIG_OF_BROKER_GROUPBY_TRIM_THRESHOLD = "pinot.broker.groupby.trim.threshold";
     public static final int DEFAULT_BROKER_GROUPBY_TRIM_THRESHOLD = 1_000_000;
+    public static final String CONFIG_OF_MIN_BROKER_GROUPBY_TRIM_SIZE = "pinot.broker.min.groupby.trim.size";
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java:270]:
```suggestion
    public static final String CONFIG_OF_BROKER_MIN_GROUP_TRIM_SIZE = "pinot.broker.min.group.trim.size";
```

**Review by Jackie-Jiang — APPROVED** [review]
LGTM

**Comment by Jackie-Jiang** [comment]
Can you help add this new config into the Pinot documentation?

**Comment by vvivekiyer** [comment]
@Jackie-Jiang  Added the new config to documentation - https://docs.pinot.apache.org/users/user-guide-query/query-syntax/grouping-algorithm

**Comment by Jackie-Jiang** [comment]
@vvivekiyer Thanks for adding the documentation! Seems we still have some configs not able to be overridden by query option, including this new added one. Do you want to help contribute the query option override for them?

**Comment by vvivekiyer** [comment]
@Jackie-Jiang sure, I'll create a PR to add queryOptions for the following configs:
1. pinot.server.query.executor.groupby.trim.threshold
2. pinot.broker.min.group.trim.size
3. pinot.server.query.executor.num.groups.limit

**Comment by vvivekiyer** [comment]
Query option override PR - https://github.com/apache/pinot/pull/11984 
cc: @Jackie-Jiang

**Pull request merged**